fs/dyndev.h

```diff
@@ -0,0 +1,31 @@
+#ifndef FS_DYN_DEV_H
+#define FS_DYN_DEV_H
+
+// dyn_dev's are dynamically added devices (character only for now)
+// with custom dev_ops assigned
+// It's useful to add new device "drivers" in runtime (for example,
+// devices only present on some platforms)
+
+// (int)('D'+'Y') == 157
+#define DYN_DEV_MAJOR 157
+
+// dev_ops handing DYN_DEV_MAJOR major number
+extern struct dev_ops dyn_dev;
+
+// Registeres new device with major number DYN_DEV_MAJOR and provided
+// minor number handled by provided ops.
+//
+// ops should be valid for "kernel" lifetime (should not be freed, but
+// might be static), and should not be null
+//
+// minor should be 0-255 or -1. if it is -1, minor number is allocated
```

Comment:
Let's require each new device to choose a fixed minor number to avoid problems with a device file not working anymore. We can have a central header file to manage the allocation (similar to linux's devices.txt).

Reply:
Should I remove dynamic minor number selection to simplify dyn_dev_register?

Reply:
Yeah that's what I was thinking.